src/routes/version.rs

```diff
@@ -0,0 +1,52 @@
+use std::env;
+
+use axum::{
+    response::{IntoResponse, Response},
+    routing::get,
+    Json, Router,
+};
+use serde::Serialize;
+
+pub fn route() -> Router {
+    Router::new()
+        .route("/version", get(version))
+        .route("/version-details", get(version_details))
+}
+
+#[derive(Serialize)]
+struct VersionDetails {
+    version: &'static str,
+    version_bits: VersionBits,
+    git_commit_hash: &'static str,
+    current_endpoint_prefix: &'static str,
+    repository: &'static str,
+}
+
+#[derive(Serialize)]
+struct VersionBits {
+    major: &'static str,
+    minor: &'static str,
+    patch: &'static str,
+}
+
+async fn version() -> Response {
+    let v = env!("CARGO_PKG_VERSION");
+
+    return v.into_response();
+}
```

Comment:
Please do not use shortened variable names like that. They are not descriptive and make the code hard to read. Variable names should be self-explanatory and require as little context as possible to be understandable.

```suggestion
async fn version() -> Response {
    let version = env!("CARGO_PKG_VERSION");

    return version.into_response();
}
```

Reply:
The effect you mention is overstated, but I guess I can just make this return the env!() explicitly. You do realise we'll be using `v` as an abbreviation as part of endpoint versioning, correct?

Reply:
Return it explicitly, then.